app/grandchallenge/challenges/models.py

```diff
@@ -1402,3 +1403,81 @@ class ChallengeRequestGroupObjectPermission(GroupObjectPermissionBase):
     content_object = models.ForeignKey(
         ChallengeRequest, on_delete=models.CASCADE
     )
+
+
+class TaskResponsibleChoices(models.TextChoices):
+    SUPPORT = "SUP", "Support"
+    CHALLENGE_ORGANIZERS = "ORG", "Challenge Organizers"
+
+
+class OnboardingTask(FieldChangeMixin, UUIDModel):
+    ResponsibleChoices = TaskResponsibleChoices
+
+    class Meta:
+        permissions = [
+            ("complete_onboaringtask", "Can mark this task as completed")
+        ]
+
+    created = models.DateTimeField(editable=False)
+    challenge = models.ForeignKey(
+        to=Challenge,
+        on_delete=models.CASCADE,
+        related_name="onboarding_tasks",
+    )
+    title = models.CharField(
+        max_length=255,
+        help_text="Title of this task",
+    )
+
+    description = models.TextField(
+        blank=True, help_text="Description of this task."
+    )
+    complete = models.BooleanField(
+        default=False,
+        help_text="If true, this task is complete.",
+    )
+    completed_at = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text="Time this task was last marked completed.",
```

Comment:
Again for my understanding: the help text suggests that we allow tasks to be completed multiple times. Is that the case, and if so, why?

Reply:
I think it wise to allow tasks to be have a toggable complete:

- Allow for accidental complete correction.
- Allow for: "no you did **not** complete this" reverting.

The help text is to indicate that it is only the last time someone marked it complete. If it would not mention it: it could also be the first time.

Reply:
Yeah ok, makes sense then. I think this should all be reserved for us staff users though.  A user should only be able to mark it as completed once.

Reply:
Why?

We could have a 'hot' period for the complete where a user is not allowed to undo the complete if it's been marked complete for more than, say, a day?

Reply:
On second thought, I guess it's ok to allow admins to uncomplete a task in case they accidentally close it. I had a workflow where it's us doing to completion marking in mind, just like with challenge requests, and then you'd want to lock it down, but never mind!